src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java

```diff
@@ -195,10 +270,21 @@ public static KNNQueryBuilder fromXContent(XContentParser parser) throws IOExcep
             }
         }
 
-        KNNQueryBuilder knnQueryBuilder = new KNNQueryBuilder(fieldName, ObjectsToFloats(vector), k, filter);
-        knnQueryBuilder.ignoreUnmapped(ignoreUnmapped);
-        knnQueryBuilder.queryName(queryName);
-        knnQueryBuilder.boost(boost);
+        if ((k != null && distance != null) || (k == null && distance == null)) {
+            throw new IllegalArgumentException("[" + NAME + "] requires either k or distance must be set");
+        }
+
+        KNNQueryBuilder knnQueryBuilder = new KNNQueryBuilder(fieldName, ObjectsToFloats(vector)).filter(filter)
+            .ignoreUnmapped(ignoreUnmapped)
+            .boost(boost)
+            .queryName(queryName);
+
+        if (k != null) {
+            knnQueryBuilder = knnQueryBuilder.k(k);
```

Comment:
i don't think you need to update . You can do like 285

Reply:
Updated